### DISCUSSION TITLE
Feature request: DNS-over-QUIC

### DISCUSSION THREAD

**Discussion by phantomcraft** [DNSCrypt/dnscrypt-proxy#1914] [Ideas]
It is already implemented in this project: https://github.com/AdguardTeam/dnsproxy

The first DNS-over-QUIC resolver is: quic://dns.adguard.com

More info about this resolver: https://adguard.com/en/blog/dns-over-quic.html

SOCKS5 proxies support UDP, so, DNS-over-QUICK would work with a uptream SOCKS5 server with UDP support.

**Comment by phantomcraft** [comment]
Other AdGuard DNS-over-QUIC servers:

quic://dns-family.adguard.com
quic://dns-unfiltered.adguard.com

**Comment by lifenjoiner** [comment]
Read https://github.com/DNSCrypt/dnscrypt-proxy/discussions?discussions_q=QUIC
